### ISSUE TITLE
string.Normalize() does not work

### ISSUE THREAD

**Issue by zeroskyx** [dotnet/runtime#48173]
### Description

The `string.Normalize()` method does not work anymore.

When using string.Normalize() to remove diacritics from a string, the following code no longer works:

```csharp
string Text = "aøeiáßouyäöüÄÖÜẞxXX";

Text = Text.Normalize(NormalizationForm.FormD);

StringBuilder TextBuilder = new StringBuilder();
foreach(char ThisChar in Text)
{
	if(CharUnicodeInfo.GetUnicodeCategory(ThisChar) != UnicodeCategory.NonSpacingMark)
		TextBuilder.Append(ThisChar);
}

string Result = TextBuilder.ToString().Normalize(NormalizationForm.FormC);
```

#### Expected outcome:

The value of `Result` is: `aoeiaouyaouAOUxXX`

#### Actual outcome:

The value of `Result` is: `aøeiáouyäöüÄÖÜxXX`

### Configuration

```
NET SDK (reflecting any global.json):
 Version:   5.0.102
 Commit:    71365b4d42

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  20.10
 OS Platform: Linux
 RID:         ubuntu.20.10-x64

Host (useful for support):
  Version: 5.0.2
  Commit:  cb5f173b96

.NET SDKs installed:
  5.0.102

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.2
  Microsoft.NETCore.App 5.0.2

```

### Regression?

This did previously work; I'm not sure on which version though. Never tested on .NET5 before.

**Comment by AraHaan** [comment]
@zeroskyx try the latest 5.0.103 SDK.

**Comment by zeroskyx** [comment]
@AraHaan thanks for the rapid response (and the streamlining). I think I found the problem. A couple of updates back I set `DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=1` because of an issue.

It looks like [Normalization.cs](https://source.dot.net/#System.Private.CoreLib/Normalization.cs,39124745e49d4135) does not perform any normalization when [GlobalizationMode.cs](https://source.dot.net/#System.Private.CoreLib/GlobalizationMode.cs,f6bbb0cce0310efc) detects that environment variable.

So no bug; thanks for the help!
